### ISSUE TITLE
copilot env init command issue - named profile vs credentials

### ISSUE THREAD

**Issue by tsingh2k15** [aws/copilot-cli#2937]
copilot version: v1.11.0 Noticing an issue when not using named profile while looking at https://aws.github.io/copilot-cli/docs/commands/env-init/

**Command** 

```
  copilot env init \
      -n someName \
      --import-vpc-id someVPCId \
      --import-private-subnets subnetIds \
      --aws-access-key-id ${env.AWS_ACCESS_KEY_ID} \
      --aws-secret-access-key ${env.AWS_SECRET_ACCESS_KEY} \
      --aws-session-token ${env.AWS_SESSION_TOKEN}"
```

**Error**

`✘ read named profiles: read AWS config file /home/ec2-user/.aws/config, you might need to run aws configure first: load ini file /home/ec2-user/.aws/config: open /home/ec2-user/.aws/config: no such file or directory`

Can I get some support on this? Am I missing something in the command? There seems to be support available per https://github.com/aws/copilot-cli/issues/1228

**Comment by iamhopaul123** [comment]
Hello @tsingh2k15 I think right now though we support using temp cred, the local config file is still required. If you create a local config file then copilot should prompt you to choose which profile you'd like to use, and then you can choose "Use temporary credentials" if you want to use temp cred.

**Comment by iamhopaul123** [comment]
Also just curious could you share with us the reason why you don't want to have an aws config file locally? Is it because you use only temp cred instead of profiles? Thank you!

**Comment by tsingh2k15** [comment]
I see @iamhopaul123 . I understand that using named profile is a better practice and I am now trying that route as well.  Though it is a bit of more work at my end to first setup a named profile on Jenkins agent my pipeline runs via my own Jenkinsfile. There is nothing configured on the box, so leveraging the credentials in environment(populated via cross account assume role for my specific usecase) is my preferred route. I was expecting that it would work, it is strange that copilot would still try to check the config file!  Although aws configure would be in place on most of the systems, but this is not the case for my scenario, considering that this is still possible, would it be okay to let the above command through as the user has provided all the inputs? In either case, I think the docs and error messages should be more clear. What do you think?

**Comment by iamhopaul123** [comment]
I agree since you are not using the profile we should make it to be optional and when we cannot detect the existence of the config file, we'll just ask for temp cred and maybe with an alert prompting them to create a profile. This would be a better user experience. Thank you bringing it up!

Also as a workaround to temporarily unblock your project, could you create a mock file under `/home/ec2-user/.aws/config` or run `aws config` before using Copilot to create any environment?

**Comment by tsingh2k15** [comment]
> I agree since you are not using the profile we should make it to be optional and when we cannot detect the existence of the config file, we'll just ask for temp cred and maybe with an alert prompting them to create a profile. This would be a better user experience. Thank you bringing it up!
> 
> Also as a workaround to temporarily unblock your project, could you create a mock file under `/home/ec2-user/.aws/config` or run `aws config` before using Copilot to create any environment?

makes sense.  I am not using temp creds route anymore. I already created a mock file with a valid profile and now using that. It appears to work for the most part for `copilot env init` command. I am facing likely a new issue now where `copilot svc init` is now failing where it cannot find Dockerfile, though Dockerfile is already there. I will look into that as separate issue as it might be a miss at my end. As fyi here is the error - `stat ./Dockerfile no such file or directory` in case you may suggest something to resolve this :) I will check which directory I am in to make sure the file is there. This is not related to the original issue I reported on `copilot env init`. Please feel free to mark the actual issue as resolved by using named profile. Thank you for your support!

**Comment by tsingh2k15** [comment]
> > I agree since you are not using the profile we should make it to be optional and when we cannot detect the existence of the config file, we'll just ask for temp cred and maybe with an alert prompting them to create a profile. This would be a better user experience. Thank you bringing it up!
> > Also as a workaround to temporarily unblock your project, could you create a mock file under `/home/ec2-user/.aws/config` or run `aws config` before using Copilot to create any environment?
> 
> makes sense. I am not using temp creds route anymore. I already created a mock file with a valid profile and now using that. It appears to work for the most part for `copilot env init` command. I am facing likely a new issue now where `copilot svc init` is now failing where it cannot find Dockerfile, though Dockerfile is already there. I will look into that as separate issue as it might be a miss at my end. As fyi here is the error - `stat ./Dockerfile no such file or directory` in case you may suggest something to resolve this :) I will check which directory I am in to make sure the file is there. This is not related to the original issue I reported on `copilot env init`. Please feel free to mark the actual issue as resolved by using named profile. Thank you for your support!

just fyi, the issue i mentioned with `svc init` is resolved. I had to checkout the code again as I was switching agents in Jenkinsfile. 

I will watch for the future enhancement in docs and/or copilot-cli behavior as we discussed in this thread.
